pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/PinotPreparedStatement.java

```diff
@@ -28,25 +28,31 @@
 import org.apache.commons.codec.binary.Hex;
 import org.apache.pinot.client.base.AbstractBasePreparedStatement;
 import org.apache.pinot.client.utils.DateTimeUtils;
+import org.apache.pinot.client.utils.DriverUtils;
 
 
 public class PinotPreparedStatement extends AbstractBasePreparedStatement {
 
   private static final String QUERY_FORMAT = "sql";
+  private static final String LIMIT_STATEMENT = "LIMIT";
   private Connection _connection;
   private org.apache.pinot.client.Connection _session;
   private ResultSetGroup _resultSetGroup;
   private PreparedStatement _preparedStatement;
   private String _query;
   private boolean _closed;
   private ResultSet _resultSet;
+  private int _maxRows = Short.MAX_VALUE;
```

Comment:
Pinot should be able to handle `LIMIT Integer.MAX_VALUE`. Can you please check if you explicitly put a threshold on broker via config `pinot.broker.query.response.limit`?

Reply:
ok. let me check. I am using the out of the box realtime quickstart. The Max limit which it was able to handle was 10^9.
Even 10^9 + 1 was giving Internal server error.

Reply:
resolved after pulling the latest fixes in master.